CONTRIBUTING.md

```diff
@@ -8,6 +6,35 @@ Complete your CLA here: <https://code.facebook.com/cla>
 
 By contributing to osquery, you agree that your contributions will be licensed under both the `LICENSE` file and the `COPYING` file in the root directory of this source tree.
 
+## Guidelines for contributing features to osquery core
+
+The software housed in this repo is known as osquery core. While there are occasional exceptions, contributions to core should abide by the following osquery guiding principles in order to be accepted:
+1. osquery doesn’t pry into user data
```

Comment:
We had a small discussion internally, I think @fmanco had some thoughts about this line around `user data` is really loosely defined, and we'd want to investigate this further -- @fmanco?

Reply:
Yeah, I understand the sentiment behind this sentence but it's going to be hard to apply or even explain without defining what user data is, e.g. is the process list user data? What about IP addresses? What about file names? It's very hard to run osquery without collecting user data actually.

I'm open for suggestions but at this point I think we could just merge this without point 1 and we could always amend that later if we find the right way.

I'm working on a rewrite of the contributing guide, clarifying some points and adding code guidelines and style guides to make contributing easier. I would love to get this merged before so that I can use it. Let me know what you guys think.

Reply:
I'll take care of it and merge it.